crates/ruff_python_ast/src/visitor/preorder.rs

```diff
@@ -7,64 +8,89 @@ use crate::{
 
 /// Visitor that traverses all nodes recursively in pre-order.
 pub trait PreorderVisitor<'a> {
+    #[inline(always)]
+    fn enter_node(&mut self, _node: AnyNodeRef<'a>) -> TraversalSignal {
+        TraversalSignal::Traverse
+    }
+
+    #[inline(always)]
+    fn leave_node(&mut self, _node: AnyNodeRef<'a>) {}
+
+    #[inline]
```

Comment:
do we have evidence that this is needed/helpful?

Reply:
It can be beneficial across crate boundaries because it tells Rust to emit the necessary inline information to inline across crate boundaries. However, this isn't relevant for us because we use LTO. 

The use case where it is useful for us is in debug builds. To my knowledge, Rust doesn't perform any inline in debug builds *except* for functions attributed with `inline`. This means, we get slightly better performance for debug builds.